oauth2client/flask_util.py

```diff
@@ -0,0 +1,402 @@
+
+from oauth2client.client import FlowExchangeError
+from oauth2client.client import OAuth2Credentials
+from oauth2client.client import OAuth2WebServerFlow
+from oauth2client.client import Storage
+from oauth2client import clientsecrets
+
+
+DEFAULT_SCOPES = ('email',)
+
+
+class OAuth2(object):
+    """Flask extension for making OAuth 2.0 easier.
+
+    Configuration values:
+        * GOOGLE_OAUTH2_CLIENT_SECRETS_JSON path to a client secrets json file,
+            obtained from the credentials screen in the Google Developers
+            console.
+        * GOOGLE_OAUTH2_CLIENT_ID the oauth2 credentials' client ID. This is
+            only needed if OAUTH2_CLIENT_SECRETS_JSON is not specified.
+        * GOOGLE_OAUTH2_CLIENT_SECRET the oauth2 credentials' client secret.
+            This is only needed if OAUTH2_CLIENT_SECRETS_JSON is not specified.
+
+    If app is specified, all arguments will be passed along to init_app.
+
+    If no app is specified, then you should call init_app in your application
+    factory to finish initialization.
+    """
+
+    def __init__(self, app=None, *args, **kwargs):
+        self.app = app
+        if app is not None:
+            self.init_app(app, *args, **kwargs)
+
+    def init_app(self, app, scopes=None, client_secrets_file=None,
+                 client_id=None, client_secret=None, authorize_callback=None,
+                 logout_callback=None, storage=None, **kwargs):
+        """Initialize this extension for the given app.
+
+        Arguments:
+            app: A Flask application.
+            scopes: Optional list of scopes to authorize.
+            client_secrets_file: Path to a file containing client secrets. You
+                can also specify the OAUTH2_CLIENT_SECRETS_JSON config value.
+            client_id: If not specifying a client secrets file, specify the
+                OAuth2 client id. You can also specify the
+                GOOGLE_OAUTH2_CLIENT_ID config value. You must also provide a
+                client secret.
+            client_secret: The OAuth2 client secret. You can also specify the
+                GOOGLE_OAUTH2_CLIENT_SECRET config value.
+            authorize_callback: A function that is executed after successful
+                user authorization.
+            logout_callback: A function that is executed after the user is
+                directed to /oauth2logout
+            storage: A oauth2client.client.Storage subclass for storing the
+                credentials. By default, this is a Flask session based storage.
+            kwargs: Any additional args are passed along to the Flow
+                constructor.
+        """
+        self.app = app
+        self.authorize_callback = authorize_callback
+        self.logout_callback = logout_callback
+        self.flow_kwargs = kwargs
+
+        if not storage:
+            storage = FlaskSessionStorage()
+        self.storage = storage
+
+        if not scopes:
```

Comment:
Checking the boolean value of a keyword argument rather than asking if it is None is always a code smell to me.

If the user explicitly passes an empty sequence you override that user-provided value - is that intended?